src/main/client/type.c

```diff
@@ -620,12 +663,30 @@ static int AerospikeClient_Type_Init(AerospikeClient *self, PyObject *args,
                 if (PyUnicode_Check(py_addr)) {
                     addr = strdup((char *)PyUnicode_AsUTF8(py_addr));
                 }
+                else {
+                    // ["hosts"][<10 chars max>][0]
+                    // 1234567890123456789012345678
+                    // 0        10        20    + 1 = 29
+                    char keys_str[29];
+                    snprintf(keys_str, 29, "[\"hosts\"][%d][0]", i);
+                    initialize_config_value_type_err(
+                        config_value_type_error_msg, keys_str, "str",
+                        &error_code);
+                    goto CONSTRUCTOR_ERROR;
+                }
+
                 py_port = PyTuple_GetItem(py_host, 1);
-                if (PyLong_Check(py_port)) {
+                if (PyLong_CheckExact(py_port)) {
                     port = (uint16_t)PyLong_AsLong(py_port);
                 }
                 else {
-                    port = 0;
+                    char keys_str[29];
+                    snprintf(keys_str, 29, "[\"hosts\"][%d][1]", i);
+                    initialize_config_value_type_err(
```

Comment:
I think the "29" should be a constant, you could move the explanation from line 667 to the constant deceleration and give it a little more context.